pkg/api/model_to_api.go

```diff
@@ -0,0 +1,116 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/internal/model"
+	"github.com/operator-framework/operator-registry/internal/property"
+)
+
+func ConvertModelBundleToAPIBundle(b model.Bundle) (*Bundle, error) {
+	props, err := parseProperties(b.Properties)
+	if err != nil {
+		return nil, fmt.Errorf("parse properties: %v", err)
+	}
+	skipRange := ""
+	if len(props.SkipRanges) > 0 {
+		skipRange = string(props.SkipRanges[0])
+	}
+	return &Bundle{
+		CsvName:      b.Name,
+		PackageName:  b.Package.Name,
+		ChannelName:  b.Channel.Name,
+		BundlePath:   b.Image,
+		ProvidedApis: gvksProvidedtoAPIGVKs(props.GVKs),
+		RequiredApis: gvksRequirestoAPIGVKs(props.GVKsRequired),
+		Version:      props.Packages[0].Version,
+		SkipRange:    skipRange,
+		Dependencies: convertModelPropertiesToAPIDependencies(b.Properties),
+		Properties:   convertModelPropertiesToAPIProperties(b.Properties),
+		Replaces:     b.Replaces,
+		Skips:        b.Skips,
+		CsvJson:      b.CsvJSON,
+		Object:       b.Objects,
+	}, nil
+}
+
+func parseProperties(in []property.Property) (*property.Properties, error) {
```

Comment:
The main reason for the difference is because `api.Bundle` has a `SkipRange` field that allows either 0 or 1 skip ranges. When converting to the `api.Bundle`, we have to verify that no more than one skip range is defined.

The model itself doesn't have this constraint built-in, which is why the declcfg_to_model function does not include this check.

Reply:
I'm a bit confused. What do you mean by `allows either 0 or 1 skip ranges`? skipRange is just a string in `api.Bundle`. You mean with `model.Bundle`, they can have multiple `skipRange` properties?

Reply:
Correct.

Reply:
Given this restriction on skipRange is universal, then I don't see the reason why we don't impose this restriction on `declcfg` side as well.

Reply:
> this restriction on skipRange is universal

I think that's the current state of the world with the current resolver and the current GRPC API. But that might not always necessarily be the case when we've updated the resolver and gotten rid of the GRPC server.

Reply:
Sure. I can see that you want to model side to be more generic and not discriminating against this. I can see a future where the api.Bundle no longer needs a specific field for `skipRange` (or no longer exists in the new non-grpc API we use) but I don't see a future where we allow multiple `skipRange, right? However, the fact that we use the model to validate config. If the config is considered valid even with multiple `skipRange`, then it will fail the serve part. That is not a good UX.

Reply:
Chatted offline with @dinhxuanvu. We'll handle this concern in #618

Reply:
Joe and I agree that we should have this "validation" on the model side so the validate and serve can be in sync in term of pass/fail on the same scenario which in case is multiple skipRange. We will keep things as they are in Joe's PR and will address this on my PR (which should be merged after Joe's).